### PR TITLE
Bug fix for clean up in the 'Forgot Passcode' use case

### PR DESCRIPTION
**Pull request by bhariharan**
A few beautiful bug fixes for our passcode users

### CODE REVIEW

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java [hunk 1]
```diff
@@ -222,27 +220,21 @@ public static synchronized void deleteDatabase(Context ctx, String dbNamePrefix,
 				StringBuffer communityDBNamePrefix = new StringBuffer(dbNamePrefix);
 				String accountSuffix = account.getUserLevelFilenameSuffix();
 				communityDBNamePrefix.append(accountSuffix);
-				
-		    	final String dbPath = ctx.getApplicationInfo().dataDir + "/databases";
```
Comment (bhariharan) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java]: Simply moved this out to its own method.

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java [hunk 2]
```diff
@@ -222,27 +220,21 @@ public static synchronized void deleteDatabase(Context ctx, String dbNamePrefix,
...
 			Log.e("DBOpenHelper:deleteDatabase", "Exception occurred while attempting to delete database.", e);
 		}
 	}
 
+	/**
+	 * Deletes all remaining databases.
+	 *
+	 * @param ctx Context.
+	 */
+	public static synchronized void deleteAllDatabases(Context ctx) {
```
Comment (bhariharan) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java]: Added a method here to delete everything.

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/app/SalesforceSDKManagerWithSmartStore.java [hunk 3]
```diff
@@ -180,6 +180,8 @@ protected void cleanUp(Activity frontActivity, Account account) {
     		if (userAccount != null && hasSmartStore(userAccount)) {
     			DBOpenHelper.deleteDatabase(getAppContext(), userAccount);
     		}
+    	} else {
```
Comment (bhariharan) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/app/SalesforceSDKManagerWithSmartStore.java]: Cleaning up everything in the ```Forgot Passcode``` use case.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java [hunk 4]
```diff
@@ -828,21 +828,43 @@ private void removeAccount(ClientManager clientMgr, final boolean showLoginPage,
...
+    	 * we don't have the passcode hash to decrypt them. Hence, we query
+    	 * AccountManager directly here and remove the accounts for the case
+    	 * where 'account == null'. If AccountManager doesn't have accounts
+    	 * either, then there's nothing to do.
+    	 */
     	if (account == null) {
-    		EventsObservable.get().notifyEvent(EventType.LogoutComplete);
-    		if (showLoginPage) {
-    			startSwitcherActivityIfRequired();
+    		final AccountManager accMgr = AccountManager.get(context);
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java]: Some magic based on my code comments above.

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java [hunk 5]
```diff
@@ -290,6 +282,23 @@ public void postKey(SQLiteDatabase database) {
 		}
 	};
 
+	private static void deleteFiles(Context ctx, String prefix) {
+		final String dbPath = ctx.getApplicationInfo().dataDir + "/databases";
+    	final File dir = new File(dbPath);
```
Comment (khawkins) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java]: Spacing seems like it went awry here.

**Comment by khawkins** [comment]
Otherwise, logic looks reasonable to me.

**Comment by wmathurin** [comment]
Looks fine.

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java (hunk 5 above)
Comment (bhariharan) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java]: It looks fine on ```Eclipse``` for some reason.

**Pull request merged**